tests/Unit/Entity/BannerSelection/CampaignTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace WMDE\BannerServer\Tests\Unit\Entity\BannerSelection;
+
+use WMDE\BannerServer\Entity\BannerSelection\Campaign;
+use WMDE\BannerServer\Entity\BannerSelection\TrafficImpressionThreshold;
+
+class CampaignTest extends \PHPUnit\Framework\TestCase {
+
+	public static function returnRandomBucket() {
+		return '';
+	}
+
+	public function testGivenTimeOutOfDateRange_campaignIsNotActive() {
+		$campaign = new Campaign(
+			'C18_WMDE_Test',
+			new \DateTime( '2018-10-01' ),
+			new \DateTime( '2018-10-31' ),
+			[],
+			new TrafficImpressionThreshold( 10, 10 )
+		);
+		$this->assertTrue(
+			$campaign->isInActiveDateRange( new \DateTime( '2018-10-20' ) ),
+			'date given is between the start and the end of the campaign'
+		);
+	}
+
+	public function testGivenTimeTheDateRange_campaignIsActive() {
+		$campaign = new Campaign(
+			'C18_WMDE_Test',
+			new \DateTime( '2018-10-01' ),
+			new \DateTime( '2018-10-31' ),
+			[],
+			new TrafficImpressionThreshold( 10, 10 )
+		);
+		$this->assertFalse(
+			$campaign->isInActiveDateRange( new \DateTime( '2018-09-22' ) ),
+			'date given is before the start of the campaign'
+		);
+	}
+
+	public function testGivenBucketId_returnsBucket() {
+		$campaign = new Campaign(
+			'C18_WMDE_Test',
+			new \DateTime( '2018-10-01' ),
+			new \DateTime( '2018-10-31' ),
+			[],
+			new TrafficImpressionThreshold( 10, 10 )
+		);
+		$this->assertEquals(
+			$campaign->selectBucket( 'C18_WMDE_Test_var', '$this->returnRandomBucket' ),
+			'C18_WMDE_Test_var'
+		);
+	}
+
+	public function testGivenNoBucketId_returnsRandomBucket() {
+		$campaign = new Campaign(
+			'C18_WMDE_Test',
+			new \DateTime( '2018-10-01' ),
+			new \DateTime( '2018-10-31' ),
+			[],
+			new TrafficImpressionThreshold( 10, 10 )
+		);
+		$this->assertEquals( $campaign->selectBucket( null, '$this->returnRandomBucket' ), '' );
```

Comment:
An anonymous function would be sufficient. 

If you want to use a public static function from this test class, pass `[CampaignTest::class, 'returnRandomBucket']` instead. Unfortunately, we have to reference methods by string, but for class names, we can always use the `::class`. This helps later, when we rename the class in PHPStiorm refactoring.